cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -593,6 +745,40 @@ struct Resources {
     stack_slots: Vec<(StackSlot, StackSize)>,
 }
 
+impl Resources {
+    /// Returns [JumpTable]'s where all blocks are forward of `block`
+    fn forward_jump_tables(&self, builder: &FunctionBuilder, block: Block) -> Vec<JumpTable> {
+        // Unlike with the blocks below jump table targets are not ordered, thus we do need
+        // to allocate a Vec here.
+        let jump_tables = &builder.func.jump_tables;
+        self.jump_tables
+            .iter()
+            .copied()
+            .filter(|jt| jump_tables[*jt].iter().all(|target| *target > block))
+            .collect()
```

Comment:
On further thought, we could avoid allocating here by sorting `self.jump_tables` in `generate_jumptables`. I'm happy to merge without it but I'm going to write down the idea in case you decide you want to do it.

The sort key is `|jt| builder.func.jump_tables[*jt].iter().min()`. In `generate_jumptables` you could use that as the argument to `self.resources.jump_tables.sort_by_cached_key(...)`. Then here you could use it in `self.jump_tables.partition_point(... <= block)`.

To keep the sort key computation in only one place, you could have `generate_jumptables` save a pair of `(min_block, table_id)` in `self.resources.jump_tables`. Then the sort is just `self.resources.jump_tables.sort_unstable_by_key(|(min_block, _)| *min_block)` and this function just needs `self.jump_tables.partition_point(|(min_block, _)| *min_block <= block)`.

Reply:
That sounds quite doable! I'm going to add a TODO comment to revisit later, I'd prefer to tackle the other fuzzgen issues for now.